app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -2,6 +2,10 @@
 # event type messages. Parsing of the message is deferred to the caller of
 # #poll.
 class ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream
+  # Topic wasn't found (event catcher likely not set up)
+  class TopicNotFound < MiqException::Error
```

Comment:
Nice!  I didn't see this exception the first time; looks like it already handles sleeping as well.  That'll make this change a lot shorter.